src/main/java/org/mockbukkit/mockbukkit/inventory/InventoryViewMock.java

```diff
@@ -145,15 +146,59 @@ public void setCursor(@Nullable ItemStack item)
 	@Override
 	public void setItem(int slot, @Nullable ItemStack item)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		int topSize = getTopInventory().getSize();
+		if (slot < 0)
+		{
+			if (item != null)
+			{
+				Location location = bottomInventory.getLocation(); // should be the player
+				if (location != null)
+				{
+					location.getWorld().dropItemNaturally(location, item);
+					return;
+				}
+			}
+
+			throw new IndexOutOfBoundsException(slot);
+		}
+
+		if (slot < topSize)
+		{
+			getTopInventory().setItem(slot, item);
+			return;
+		}
+
+		slot -= topSize;
+		if (slot < getBottomInventory().getSize())
+		{
+			getBottomInventory().setItem(slot, item);
+			return;
+		}
+
+		throw new IndexOutOfBoundsException(slot);
 	}
 
 	@Override
 	public @Nullable ItemStack getItem(int slot)
```

Comment:
Just going to note that the item should not be copied here, this implementation is fine